cupy/creation/matrix.py

```diff
@@ -22,17 +22,17 @@ def diag(v, k=0):
     .. seealso:: :func:`numpy.diag`
 
     """
-    if isinstance(v, cupy.ndarray):
-        if v.ndim == 1:
-            size = v.size + abs(k)
-            ret = cupy.zeros((size, size), dtype=v.dtype)
-            ret.diagonal(k)[:] = v
-            return ret
-        else:
-            return v.diagonal(k)
-    else:
+    if numpy.isscalar(v):
         return cupy.array(numpy.diag(v, k))
 
+    if v.ndim == 1:
```

Comment:
How about `if numpy.isscalar(v) or v.ndim == 1:` ?

Reply:
`v` may be a Python scalar (c.f., `np.isscalar(3)` returns `True`).

Reply:
OK.